snippets/rust.snippets

```diff
@@ -93,70 +108,72 @@ snippet loop "loop {}" b
 	loop {
 		${0}
 	}
-snippet while
+snippet while "while loop"
 	while ${1:condition} {
 		${0}
 	}
-snippet for
+snippet for "for ... in ... loop"
 	for ${1:i} in ${2:0u..10} {
 		${0}
 	}
-snippet spawn
-	spawn(proc() {
-		${0}
-	});
-snippet chan
-	let (${1:tx}, ${2:rx}): (Sender<${3:int}>, Receiver<${4:int}>) = channel();
-snippet duplex
-	let (${1:from_child}, ${2:to_child}) = sync::duplex();
 # TODO commenting
-snippet todo
+snippet todo "TODO comment"
 	// [TODO]: ${0:Description}
 snippet fixme "FIXME comment"
 	// FIXME: $0
 
 # Struct
-snippet st
+snippet st "Struct definition"
 	struct ${1:`substitute(vim_snippets#Filename(), '\(_\|^\)\(.\)', '\u\2', 'g')`} {
 		${0}
 	}
-snippet stn
-	struct ${1:`substitute(vim_snippets#Filename(), '\(_\|^\)\(.\)', '\u\2', 'g')`} {
+snippet impl "Struct/Trait implementation"
+	impl ${1:Type/Trait}${2: ${3:for Type}} {
+		${0}
+	}
+snippet stn "Struct with new constructor"
+	pub struct ${1:`substitute(vim_snippets#Filename(), '\(_\|^\)\(.\)', '\u\2', 'g')`} {
 		${0}
 	}
 
 	impl $1 {
 		pub fn new(${2}) -> $1 {
-			${4}return $1 {
-				${5}
-			};
+			$1 { ${3} };
```

Comment:
Explicit `return` is not idiomatic in Rust unless it's early, see [examples of this construct][1]. The tabstop at beginning of line to allow adding additional lines here was awkward to me and not very useful.

[1]: http://doc.rust-lang.org/book/method-syntax.html#static-methods